hypervisor/src/lib.rs

```diff
@@ -187,3 +188,98 @@ pub enum IrqRoutingEntry {
     #[cfg(feature = "mshv")]
     Mshv(mshv_bindings::mshv_msi_routing_entry),
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum StandardRegisters {
+    #[cfg(feature = "kvm")]
+    Kvm(kvm_bindings::kvm_regs),
+    #[cfg(all(feature = "mshv", target_arch = "x86_64"))]
+    Mshv(mshv_bindings::StandardRegisters),
+}
+
+impl Default for StandardRegisters {
+    fn default() -> Self {
+        #[cfg(feature = "kvm")]
+        return kvm_bindings::kvm_regs::default().into();
+        #[cfg(feature = "mshv")]
+        #[allow(unreachable_code)]
+        {
+            mshv_bindings::StandardRegisters::default().into()
+        }
+    }
+}
+
+macro_rules! set_x86_64_reg {
+    ($reg_name:ident) => {
+        concat_idents!(method_name = "set_", $reg_name {
+            #[cfg(target_arch = "x86_64")]
+            impl StandardRegisters {
+                pub fn method_name(&mut self, val: u64) {
+                    match self {
+                        #[cfg(feature = "kvm")]
+                        StandardRegisters::Kvm(s) => s.$reg_name = val,
+                        #[cfg(feature = "mshv")]
+                        StandardRegisters::Mshv(s) => s.$reg_name = val,
+                    }
+                }
+            }
+        });
+    }
+}
+
+macro_rules! get_x86_64_reg {
+    ($reg_name:ident) => {
+        concat_idents!(method_name = "get_", $reg_name {
+            #[cfg(target_arch = "x86_64")]
+            impl StandardRegisters {
+                pub fn method_name(&self) -> u64 {
+                    match self {
+                        #[cfg(feature = "kvm")]
+                        StandardRegisters::Kvm(s) => s.$reg_name,
+                        #[cfg(feature = "mshv")]
+                        StandardRegisters::Mshv(s) => s.$reg_name,
+                    }
+                }
+            }
+        });
+    }
+}
+
+
+set_x86_64_reg!(rax);
+set_x86_64_reg!(rbx);
+set_x86_64_reg!(rcx);
+set_x86_64_reg!(rdx);
+set_x86_64_reg!(rsi);
+set_x86_64_reg!(rdi);
+set_x86_64_reg!(rsp);
+set_x86_64_reg!(rbp);
+set_x86_64_reg!(r8);
+set_x86_64_reg!(r9);
+set_x86_64_reg!(r10);
+set_x86_64_reg!(r11);
+set_x86_64_reg!(r12);
+set_x86_64_reg!(r13);
+set_x86_64_reg!(r14);
+set_x86_64_reg!(r15);
+set_x86_64_reg!(rip);
+set_x86_64_reg!(rflags);
+
+get_x86_64_reg!(rax);
+get_x86_64_reg!(rbx);
+get_x86_64_reg!(rcx);
+get_x86_64_reg!(rdx);
+get_x86_64_reg!(rsi);
+get_x86_64_reg!(rdi);
+get_x86_64_reg!(rsp);
+get_x86_64_reg!(rbp);
+get_x86_64_reg!(r8);
+get_x86_64_reg!(r9);
+get_x86_64_reg!(r10);
+get_x86_64_reg!(r11);
+get_x86_64_reg!(r12);
+get_x86_64_reg!(r13);
+get_x86_64_reg!(r14);
+get_x86_64_reg!(r15);
+get_x86_64_reg!(rip);
+get_x86_64_reg!(rflags);
```

Comment:
I'd suggest to have a separate commit to define these getter/setter and explain why they will be needed.

Reply:
Done.